storage: TestMVCCIterateTimeBound failed

storage.TestMVCCIterateTimeBound [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11806628?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11806628?buildTab=artifacts#/) on master @ [985662236d7bf273b93a7b5e32def8e2d1043640](https://github.com/cockroachdb/cockroach/commits/985662236d7bf273b93a7b5e32def8e2d1043640):


```
=== RUN   TestMVCCIterateTimeBound
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/c5eb8fc8b8e683c19f3c3e4238f64094/logTestMVCCIterateTimeBound2030547181
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestMVCCIterateTimeBound (0.20s)
=== RUN   TestMVCCIterateTimeBound/0.000000019,0-0.000000020,0
    mvcc_incremental_iterator_test.go:1525: source of truth had no expected KVs; likely a bug in the test itself
    --- FAIL: TestMVCCIterateTimeBound/0.000000019,0-0.000000020,0 (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCIterateTimeBound.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31603

--- a/pkg/storage/bench_test.go
+++ b/pkg/storage/bench_test.go
@@ -720,7 +720,7 @@ func loadTestData(dir string, numKeys, numBatches, batchTimeSpan, valueBytes int
 			batch = eng.NewBatch()
 			minWallTime = sstTimestamps[i/batchSize]
 		}
-		timestamp := hlc.Timestamp{WallTime: minWallTime + rand.Int63n(int64(batchTimeSpan))}
+		timestamp := hlc.Timestamp{WallTime: minWallTime + rng.Int63n(int64(batchTimeSpan))}
 		value := roachpb.MakeValueFromBytes(randutil.RandBytes(rng, valueBytes))
 		value.InitChecksum(key)
 		if err := MVCCPut(ctx, batch, key, timestamp, value, MVCCWriteOptions{}); err != nil {

--- a/pkg/storage/mvcc_incremental_iterator_test.go
+++ b/pkg/storage/mvcc_incremental_iterator_test.go
@@ -1463,6 +1463,10 @@ func TestMVCCIterateTimeBound(t *testing.T) {
 	}
 	defer eng.Close()
 
+	// TODO(jackson): This test is tightly coupled with the prng seed constant
+	// in loadTestData. Some of the assertions below assume there exist keys
+	// within narrow bounds (eg, [19,20)). This is the case today for the
+	// current prng seed, but it's a bit too brittle.
 	for _, testCase := range []struct {
 		start hlc.Timestamp
 		end   hlc.Timestamp

